### PR TITLE
BICAWS-2753: Add heading and summary box to case details page

### PR DESCRIPTION
**Pull request by agnieszkazbikowska-madetech**
[BICAWS-2753](https://dsdmoj.atlassian.net/jira/software/projects/BICAWS/boards/522?selectedIssue=BICAWS-2753)

- create a summary box for case details,
- create an urgent badge component & corresponding storybook,
- update cy tests for case details, index,
- update storybook for court case details.

[BICAWS-2753]: https://dsdmoj.atlassian.net/browse/BICAWS-2753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx [hunk 1]
```diff
@@ -37,17 +40,17 @@ export const DetailsNotLockedByAnotherUser: ComponentStory<typeof CourtCaseDetai
 
 DetailsNotLockedByAnotherUser.play = ({ canvasElement }) => {
   const canvas = within(canvasElement)
-  expect(canvas.getByText(courtCase.ptiurn)).toBeInTheDocument()
-  expect(canvas.getByText(courtCase.courtName)).toBeInTheDocument()
+  expect(canvas.getAllByText(courtCase.ptiurn)).toHaveLength(2)
```
Comment (C-gyorfi) [inline src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx:43]: We could remove these fields from the old placeholder table so its only show in the summary box at top of the page:

image

File: test/test-data/error_list_aho.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
...
...al for Sentence on Adjournment</ds:ResultVariableText>\n<ds:ResultHalfLifeHours>72</ds:ResultHalfLifeHours>\n<br7:PNCDisposalType>4023</br7:PNCDisposalType>\n<br7:ResultClass>Adjournment post Judgement</br7:ResultClass>\n<br7:ReasonForOffenceBailConditions>to ensure appearance;</br7:ReasonForOffenceBailConditions>\n<br7:ConvictingCourt>2723</br7:ConvictingCourt>\n</br7:Result>\n</br7:Offence>\n</br7:HearingDefendant>\n</br7:Case>\n</br7:HearingOutcome>\n<br7:HasError>true</br7:HasError>\n<CXE01>\n<FSC FSCode=\"04CA\" IntfcUpdateType=\"K\"/>\n<IDS CRONumber=\"\" Checkname=\"MIDDLETONN\" IntfcUpdateType=\"K\" PNCID=\"2000/0459646G\"/>\n<CourtCases>\n<CourtCase>\n<CCR CourtCaseRefNo=\"97/1626/008395Q\" CrimeOffenceRefNo=\"\" IntfcUpdateType=\"K\"/>\n<Offences>\n<Offence>\n<COF ACPOOffenceCode=\"12:15:24:1\" CJSOffenceCode=\"TH68006\" IntfcUpdateType=\"K\" OffEndDate=\"\" OffEndTime=\"\" OffStartDate=\"28112010\" OffStartTime=\"2220\" OffenceQualifier1=\"\" OffenceQualifier2=\"\" OffenceTitle=\"Theft of pedal cycle\" ReferenceNumber=\"001\"/>\n</Offence>\n</Offences>\n</CourtCase>\n</CourtCases>\n</CXE01>\n<br7:PNCQueryDate>2022-07-27</br7:PNCQueryDate>\n</br7:AnnotatedHearingOutcome>"
```
[the remark this answers is not in the document]
Comment (agnieszkazbikowska-madetech) [reply test/test-data/error_list_aho.json:2]: I've added `PNCIdentifier>2706/1234567P` for the storybook. I'm not sure why it's highlighted the whole section...

File: src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx [hunk 3]
```diff
@@ -15,7 +15,10 @@ export default {
 } as ComponentMeta<typeof CourtCaseDetails>
 
 const courtCase = {
+  courtReference: "01NC8114723",
+  asn: "2301MB000OV",
   courtName: "Magistrates' Courts Essex Basildon",
+  courtCode: "B01OP54",
```
[the remark this answers is not in the document]
Comment (agnieszkazbikowska-madetech) [reply src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx:21]: This is a good idea! I'll do it now.

[the remark this answers is not in the document]
Comment (agnieszkazbikowska-madetech) [reply src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx:21]: All done!

File: src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx (hunk 1 above)
Comment (agnieszkazbikowska-madetech) [reply src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx]: All done!

File: src/features/CourtCaseDetails/CourtCaseDetails.tsx [hunk 4]
```diff
@@ -17,9 +18,21 @@ interface Props {
 
 const CourtCaseDetails: React.FC<Props> = ({ courtCase, aho, lockedByAnotherUser, triggersVisible }) => (
   <>
-    <Heading as="h2" size="LARGE">
-      {"Case Details"}
+    <Heading as="h2" size="LARGE" className="govuk-!-font-weight-regular">
+      {"Case details"}
```
[the remark this answers is not in the document]
Comment (agnieszkazbikowska-madetech) [reply src/features/CourtCaseDetails/CourtCaseDetails.tsx:22]: I've changed `Case details` to `h1` & loads of tests started failing, so I've looked carefully at the code & `h1` is already there - `Case Details | Bichard7`. So I left it like that. If you'd like, we can look at this together on Monday when you're not on the learning day anymore.

I also wasn't sure about the font. Is it possible that we're missing classes or something as it's a new page?

**Review by monotypical — APPROVED** [review]

**Pull request merged**